kratos/constraints/slip_constraint.cpp

```diff
@@ -0,0 +1,169 @@
+//    |  /           |
```

Comment:
ok, i did the splitting since it was actually less work to do it then to discuss further. (also admittedly, my concern was only about a coding taste, i see the advantage of having it splitted)